integration-tests/lts/insert.test.ts

```diff
@@ -140,6 +140,42 @@ describe("insert", () => {
     tc.assert<tc.IsExact<typeof r3, { id: string }[]>>(true);
   });
 
+  test("with wrapping insert .unlessConflict()", async () => {
+    const dep = e
+      .insert(e.Person, {
+        name: "dependency",
+      })
+      .unlessConflict((person) => ({
+        on: person.name,
+        else: person,
+      }));
+
+    const query = e.with(
+      [dep], // dependency used inside the insert expression
+      e
+        .insert(e.Movie, {
+          title: "WithUnlessConflict – test",
+          characters: e.select(dep, () => ({
+            "@character_name": "Peter Quill",
+          })),
+        })
+        .unlessConflict((movie) => ({
+          on: movie.title,
+        })),
```

Comment:
I think we can just make this:

```suggestion
      e.select(e.int16(42)),
```

 That'll test that it creates a `with` expression for your `unlessConflict` flavored `insert` where that dependency is _not_ in the returned expression. That's the point of explicit `with` blocks, typically. It's usually that, or if you need to move the automatic with extraction scope to a different scope in a query.

Reply:
@scotttrinh ok thanks I committed your suggestion. i don't fully follow why the previous was invalid, but i'm okay with that being out of scope lol